[BUG] Unexpected fuzzing error with "API accepted schema-violating request" on Authorization

### Checklist

- [x] I checked the [FAQ section](https://schemathesis.readthedocs.io/en/stable/faq/) of the documentation
- [x] I looked for similar issues in the [issue tracker](https://github.com/schemathesis/schemathesis/issues)
- [x] I am using the latest version of Schemathesis

### Describe the bug

Schemathesis sends a valid request that is accepted, but the test fails with "API accepted schema-violating request".

Note: this seems to be a regression in v4.5.0 as tests were OK with schemathesis up to v4.4.4 and are broken since v4.5.0 (including v4.6.0).

Error reported by schemathesis:

```
=================================== FAILURES ===================================
__________________________________ POST /ping __________________________________
1. Test Case ID: LMlBmG
- API accepted schema-violating request
    Invalid data should have been rejected
    Expected: 400, 401, 403, 404, 406, 422, 428, 5xx
    Invalid component: parameter `Authorization` - required property removed
[204] No Content:
    <EMPTY>
Reproduce with: 
    curl -X POST -H '5;' -H 'Authorization: Basic cGxhaW46dGVzdA==' http://localhost:9000/api/ping
```

The authorization provided in curl request is OK (`plain:test`), so the error "Invalid component: parameter `Authorization` - required property removed" is wrong.

### To Reproduce

🚨 **Mandatory** 🚨: Steps to reproduce the behavior:

1. Clone the weechat repository: https://github.com/weechat/weechat
2. Compile WeeChat and run with this command:

```
weechat --dir /tmp/weechat-test-api --run-command '/set relay.network.password "test"' --run-command '/set relay.network.password_hash_iterations 100' --run-command '/set relay.network.max_clients 0' --run-command '/relay add api 9000'
```

3. Run schemathesis in another terminal, in root of git repository:

```
RELAY_PASSWORD=test schemathesis run --url http://localhost:9000/api src/plugins/relay/api/weechat-relay-api.yaml
```

Schema: https://github.com/weechat/weechat/blob/main/src/plugins/relay/api/weechat-relay-api.yaml

### Expected behavior

No errors reported.

### Environment

```
- OS: Linux Debian Sid
- Python version: 3.13.9
- Schemathesis version: 4.6.0
- Spec version: Open API 3.1.0
```

### Additional context

CI example showing the issue: https://github.com/weechat/weechat/actions/runs/19612279247/job/56159854645

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,7 +4,8 @@
 
 ### :bug: Fixed
 
-  - False positive in `negative_data_rejection` check for `text/plain` and `application/octet-stream` endpoints where non-string values become valid after serialization.
+- False positive in `negative_data_rejection` check for `text/plain` and `application/octet-stream` endpoints where non-string values become valid after serialization.
+- Authentication headers are no longer re-applied during negative testing when intentionally removed. [#3355](https://github.com/schemathesis/schemathesis/issues/3355)
 
 ## [4.6.0](https://github.com/schemathesis/schemathesis/compare/v4.5.4...v4.6.0) - 2025-11-22
 

--- a/src/schemathesis/auths.py
+++ b/src/schemathesis/auths.py
@@ -18,8 +18,10 @@ from typing import (
 
 from schemathesis.core.errors import AuthenticationError, IncorrectUsage
 from schemathesis.core.marks import Mark
+from schemathesis.core.parameters import ParameterLocation
 from schemathesis.filters import FilterSet, FilterValue, MatcherFunc, attach_filter_chain
 from schemathesis.generation.case import Case
+from schemathesis.generation.meta import CoveragePhaseData, FuzzingPhaseData, StatefulPhaseData
 
 if TYPE_CHECKING:
     import requests.auth
@@ -462,6 +464,19 @@ def apply_basic_auth(case: Case, username: str, password: str) -> None:
     case._has_explicit_auth = True
 
 
+def _should_skip_auth_for_negative_testing(case: Case, param_name: str, param_location: ParameterLocation) -> bool:
+    """Check if auth should be skipped because the parameter was intentionally removed for testing."""
+    meta = case.meta
+    if not meta or not meta.generation.mode.is_negative:
+        return False
+
+    phase_data = meta.phase.data
+    if not isinstance(phase_data, (FuzzingPhaseData, CoveragePhaseData, StatefulPhaseData)):
+        return False
+
+    return phase_data.parameter == param_name and phase_data.parameter_location == param_location
+
+
 def set_on_case(case: Case, context: AuthContext, auth_storage: AuthStorage | None) -> None:
     """Set authentication data on this case.
 
@@ -482,7 +497,9 @@ def set_on_case(case: Case, context: AuthContext, auth_storage: AuthStorage | No
     # 2. Generic auth (CLI overrides or config - applies to all operations)
     basic_auth = case.operation.schema.config.auth_for(operation=case.operation)
     if basic_auth is not None:
-        apply_basic_auth(case, *basic_auth)
+        # Don't apply auth if Authorization was intentionally removed for negative testing
+        if not _should_skip_auth_for_negative_testing(case, "Authorization", ParameterLocation.HEADER):
+            apply_basic_auth(case, *basic_auth)
         return
 
     if case.operation.schema.auth.is_defined:

--- a/src/schemathesis/specs/openapi/_hypothesis.py
+++ b/src/schemathesis/specs/openapi/_hypothesis.py
@@ -232,9 +232,9 @@ def openapi_cases(
             metadata = negated_container.meta
             location_map = {
                 "query": ParameterLocation.QUERY,
-                "path_parameters": ParameterLocation.PATH,
-                "headers": ParameterLocation.HEADER,
-                "cookies": ParameterLocation.COOKIE,
+                "path": ParameterLocation.PATH,
+                "header": ParameterLocation.HEADER,
+                "cookie": ParameterLocation.COOKIE,
                 "body": ParameterLocation.BODY,
             }
             parameter_location = location_map.get(negated_container.location)

--- a/src/schemathesis/specs/openapi/adapter/security.py
+++ b/src/schemathesis/specs/openapi/adapter/security.py
@@ -5,6 +5,8 @@ from dataclasses import dataclass
 from typing import TYPE_CHECKING, Any
 
 from schemathesis.config import ApiKeyAuthConfig, HttpBasicAuthConfig, HttpBearerAuthConfig
+from schemathesis.core.parameters import ParameterLocation
+from schemathesis.generation.meta import CoveragePhaseData, FuzzingPhaseData, StatefulPhaseData
 from schemathesis.specs.openapi.auths import ApiKeyAuthProvider, HttpBasicAuthProvider, HttpBearerAuthProvider
 
 if TYPE_CHECKING:
@@ -14,6 +16,29 @@ if TYPE_CHECKING:
     from schemathesis.specs.openapi.adapter.protocol import SpecificationAdapter
 
 ORIGINAL_SECURITY_TYPE_KEY = "x-original-security-type"
+
+
+def _matches_security_parameter(
+    definition: Mapping[str, Any],
+    param_name: str,
+    param_location: ParameterLocation,
+) -> bool:
+    """Check if security definition would set the given parameter."""
+    ty = definition.get("type")
+
+    if ty == "http":
+        return param_name == "Authorization" and param_location == ParameterLocation.HEADER
+
+    if ty == "apiKey":
+        location_map = {
+            "header": ParameterLocation.HEADER,
+            "query": ParameterLocation.QUERY,
+            "cookie": ParameterLocation.COOKIE,
+        }
+        loc = definition.get("in")
+        return isinstance(loc, str) and param_name == definition.get("name") and param_location == location_map.get(loc)
+
+    return False
 
 
 class OpenApiSecurity:
@@ -70,6 +95,21 @@ class OpenApiSecurity:
             configured_schemes: Dict of configured auth schemes from config
 
         """
+        # Check if a security parameter was intentionally removed during negative testing
+        meta = case.meta
+        if meta and meta.generation.mode.is_negative:
+            phase_data = meta.phase.data
+            if isinstance(phase_data, (FuzzingPhaseData, CoveragePhaseData, StatefulPhaseData)):
+                mutated_param = phase_data.parameter
+                mutated_location = phase_data.parameter_location
+                if mutated_param and mutated_location:
+                    # Check if any security scheme would set this parameter
+                    security_definitions = self.security_definitions
+                    for definition in security_definitions.values():
+                        if _matches_security_parameter(definition, mutated_param, mutated_location):
+                            # Don't re-apply auth that was intentionally removed for testing
+                            return False
+
         # Get security requirements for this operation
         operation_definition = case.operation.definition.raw
 

--- a/src/schemathesis/specs/openapi/negative/mutations.py
+++ b/src/schemathesis/specs/openapi/negative/mutations.py
@@ -559,8 +559,11 @@ def negate_constraints(
     if is_negated:
         # Build concise description from negated constraints
         descriptions = []
+        parameter = None
         for key in negated_keys:
             value = copied[key]
+            if key == "required" and len(value) == 1:
+                parameter = value[0]
             # Special case: format required properties list nicely with quoted names
             if key == "required" and isinstance(value, list) and len(value) <= 3:
                 props = ", ".join(f"`{prop}`" for prop in value)
@@ -571,7 +574,7 @@ def negate_constraints(
 
         constraint_desc = ", ".join(descriptions)
         metadata = MutationMetadata(
-            parameter=None,
+            parameter=parameter,
             description=f"Violates {constraint_desc}",
             location=None,
         )

--- a/src/schemathesis/transport/prepare.py
+++ b/src/schemathesis/transport/prepare.py
@@ -11,7 +11,7 @@ from schemathesis.core.errors import InvalidSchema
 from schemathesis.core.output.sanitization import sanitize_url, sanitize_value
 from schemathesis.core.parameters import ParameterLocation
 from schemathesis.core.transport import USER_AGENT
-from schemathesis.generation.meta import CoveragePhaseData, CoverageScenario
+from schemathesis.generation.meta import CoveragePhaseData, CoverageScenario, FuzzingPhaseData, StatefulPhaseData
 
 if TYPE_CHECKING:
     from requests import PreparedRequest
@@ -40,14 +40,29 @@ def prepare_headers(case: Case, headers: dict[str, str] | None = None) -> CaseIn
 
 
 def get_exclude_headers(case: Case) -> list[str]:
+    if case.meta is None:
+        return []
+
+    phase_data = case.meta.phase.data
+
+    # Exclude headers that are intentionally missing
+
     if (
-        case.meta is not None
-        and isinstance(case.meta.phase.data, CoveragePhaseData)
-        and case.meta.phase.data.scenario == CoverageScenario.MISSING_PARAMETER
-        and case.meta.phase.data.parameter_location == ParameterLocation.HEADER
-        and case.meta.phase.data.parameter is not None
+        isinstance(phase_data, CoveragePhaseData)
+        and phase_data.scenario == CoverageScenario.MISSING_PARAMETER
+        and phase_data.parameter_location == ParameterLocation.HEADER
+        and phase_data.parameter is not None
     ):
-        return [case.meta.phase.data.parameter]
+        return [phase_data.parameter]
+
+    if (
+        isinstance(phase_data, (FuzzingPhaseData, StatefulPhaseData))
+        and case.meta.generation.mode.is_negative
+        and phase_data.parameter_location == ParameterLocation.HEADER
+        and phase_data.parameter is not None
+    ):
+        return [phase_data.parameter]
+
     return []
 
 

--- a/test/cli/test_negative_metadata.py
+++ b/test/cli/test_negative_metadata.py
@@ -1,5 +1,5 @@
 import pytest
-from flask import Flask, jsonify
+from flask import Flask, jsonify, request
 
 
 @pytest.mark.snapshot(replace_reproduce_with=True)
@@ -189,3 +189,61 @@ def test_text_plain_with_query_negative_still_fails(ctx, app_runner, cli, snapsh
         )
         == snapshot_cli
     )
+
+
+@pytest.mark.parametrize(
+    "auth_config",
+    [
+        {"auth": {"openapi": {"password": {"username": "plain", "password": "test"}}}},
+        {"auth": {"basic": {"username": "plain", "password": "test"}}},
+        None,
+    ],
+    ids=["openapi-auth", "basic-auth", "no-auth"],
+)
+@pytest.mark.snapshot(replace_reproduce_with=True)
+def test_removed_auth_parameter_not_reapplied(ctx, app_runner, cli, snapshot_cli, auth_config):
+    raw_schema = ctx.openapi.build_schema(
+        {
+            "/ping": {
+                "post": {
+                    "security": [{"password": []}],
+                    "responses": {"204": {"description": "No Content"}},
+                }
+            }
+        },
+        components={
+            "securitySchemes": {
+                "password": {
+                    "type": "http",
+                    "scheme": "basic",
+                }
+            }
+        },
+    )
+
+    app = Flask(__name__)
+
+    @app.route("/openapi.json")
+    def schema():
+        return jsonify(raw_schema)
+
+    @app.route("/ping", methods=["POST"])
+    def ping():
+        auth_header = request.headers.get("Authorization")
+        if not auth_header:
+            return "", 401
+        return "", 204
+
+    port = app_runner.run_flask_app(app)
+
+    args = [
+        f"http://127.0.0.1:{port}/openapi.json",
+        "--checks=negative_data_rejection",
+        "--mode=negative",
+        "--phases=fuzzing",
+        "--max-examples=5",
+        "--seed=42",
+    ]
+    kwargs = {"config": auth_config} if auth_config else {}
+
+    assert cli.run(*args, **kwargs) == snapshot_cli

